linux/system/repo.sls

```diff
@@ -142,15 +142,17 @@ linux_repo_{{ name }}:
 linux_repo_{{ name }}_key:
   cmd.run:
     - name: "echo '{{ repo.key }}' | apt-key add -"
-    - onchange:
+    - unless: "apt-key finger --with-colons | grep -qF $(echo '{{ repo-key }} | gpg --with-fingerprint --with-colons | grep -E '^fpr')"
```

Comment:
I know this is the right way, but a simpler (and usually less likely to fail) solution would be to check for the existence of </etc/apt/sources.list.d/{{ name }}.list>. This file should only be created after the GPG is imported, so the check should be safe.